third_party/amd/lib/TritonAMDGPUToLLVM/Utility.cpp

```diff
@@ -356,6 +356,137 @@ void llStore(RewriterBase &rewriter, Location loc, Value ptr, Value val,
+  if (isPredicatedLoadCA(callOp))
+    return std::make_pair(false, false);
+  if (isPredicatedLoadCG(callOp))
+    return std::make_pair(false, true);
+  if (isPredicatedLoadCV(callOp))
+    return std::make_pair(true, true);
+
+  if (isPredicatedStoreCG(callOp))
+    return std::make_pair(false, false);
+  if (isPredicatedStoreCS(callOp))
+    return std::make_pair(false, true);
+  if (isPredicatedStoreWT(callOp))
+    return std::make_pair(true, true);
+  // unsupported modifier
+  return std::make_pair(false, false);
+}
+
+// Create the auxiliary/cachepolicy value of ROCDL::RawPtrBufferLoad/StoreOp
+//   gfx942: bit 0 = sc0, bit 1 = nt, bit 3 = swz, bit 4 = sc1
+// GFX942 Vector Memory instructions (Flat, Global, Scratch, and Buffer) have 3
+// bits to control scope and cacheability:
+// - SC[1:0] System Cache level: 0=wave, 1=group, 2=device, 3=system
+// - NT Non-Temporal: 0=expect temporal reuse; 1=do not expect temporal reuse
+//
+// -----+-----+-----+-----+----+--
+// Op   | cm  | SC1 | SC0 | NT |
+// -----+-----+-----+-----+----+--
+// Load | .ca |  0  |  0  | 0  |
+//      | .cg |  0  |  1  | 1  |
+//      | .cs |  0  |  1  | 1  |
+//      | .cv |  1  |  1  | x  |
+// -----+-----+-----+-----+----+--
+// Store| .wb |  0  |  0  | 0  |
+//      | .cg |  0  |  0  | 0  |
+//      | .cs |  0  |  1  | 1  |
+//      | .wt |  1  |  x  | x  |
+// -----+-----+-----+-----+----+--
+static int32_t getCtrlBitsForCacheModifierOnGFX942(triton::CacheModifier cm,
+                                                   bool isBufferLoad) {
+  const int sc0Bit = 0b1, ntBit = 0b10, sc1Bit = 0b1000;
+  int32_t aux = 0;
+  switch (cm) {
+  case triton::CacheModifier::CA:
+    aux = 0;
+    break;
+  case triton::CacheModifier::CG:
+    if (isBufferLoad)
+      aux |= sc0Bit | ntBit;
+    break;
+  case triton::CacheModifier::CS:
+    aux |= sc0Bit | ntBit;
+    break;
+  case triton::CacheModifier::CV:
+    aux |= sc0Bit | sc1Bit;
+    break;
+  case triton::CacheModifier::WB:
+    aux |= sc1Bit;
+    break;
+  case triton::CacheModifier::WT:
+    aux |= 17;
+    break;
+  default:
+    aux = 0;
+  }
+  return aux;
+}
+
+static int32_t getDefaultCtrlBitsForCacheModifier(triton::CacheModifier cm) {
+  return 0;
+}
+
+// Cache modifiers changes how data is managed in the GPU's cache hierarchy:
+// .ca: cache at all levels with LRU policy
+// .cg: cache at L2, can use .ca or .cs
+// .cs: cache streaming, use data once
+// .cv: don't cache and fetch again
+// .wb: write-back, writes back data at all cache levels
+// .wt: write-through, write data directly to system memory
+int32_t
+getCtrlBitsForCacheModifierOnTarget(triton::CacheModifier cm, bool isBufferLoad,
+                                    mlir::triton::AMD::TargetInfo targetInfo) {
+  if (targetInfo.getGPUKind() == llvm::AMDGPU::GK_GFX942) // gfx942
+    return getCtrlBitsForCacheModifierOnGFX942(cm, isBufferLoad);
+  else
+    return getDefaultCtrlBitsForCacheModifier(cm);
```

Comment:
Should we also emit a warning here that the user is requesting cache control which Triton isn't able to provide for the target architecture?